### DISCUSSION TITLE
Understanding Normalization in inference

### DISCUSSION THREAD

**Discussion by jaideep11061982** [ultralytics/yolov5#4044] [Q&A]
Can some one help understand why do we  use 1 and 0 order   inside  detect.py   for gn , latter this gn is used to divide xywh  bbox coordinates . 
what should be done  during Denorm  same process ?

```
for i, det in enumerate(pred):  # detections per image
            if webcam:  # batch_size >= 1
                p, s, im0, frame = path[i], f'{i}: ', im0s[i].copy(), dataset.count
            else:
                p, s, im0, frame = path, '', im0s.copy(), getattr(dataset, 'frame', 0)

            p = Path(p)  # to Path
            save_path = str(save_dir / p.name)  # img.jpg
            txt_path = str(save_dir / 'labels' / p.stem) + ('' if dataset.mode == 'image' else f'_{frame}')  # img.txt
            s += '%gx%g ' % img.shape[2:]  # print string
            **gn = torch.tensor(im0.shape)[[1, 0, 1, 0]]  # normalization gain whwh**
```

```
 for *xyxy, conf, cls in reversed(det):
                    if save_txt:  # Write to file
                        xywh = (xyxy2xywh(torch.tensor(xyxy).view(1, 4)) / gn).view(-1).tolist()  # normalized xywh
```

**Comment by glenn-jocher** [comment]
gn contains [width, height, width, height] of image.

**Reply by jaideep11061982** [reply]
ok
